packages/hooks/src/useBool/useBool.ts

```diff
@@ -0,0 +1,28 @@
+import { useCallback, useState } from "react";
+
+declare const brand: unique symbol;
+
+type Callback = () => void;
+
+export type SetTrue = Callback & {
+  [brand]: "SetTrue";
+};
```

Comment:
I've never seen this style of declaration. Any chance you can explain the rationale?

Reply:
In this particular case it's used to make the type opaque so that when inspecting a value of this type (like hovering over it in the code editor) it displays `SetTrue` rather than  `() =>void`

This strategy is also called branding sometimes. It allows you to add specificity to primitive types. Often useful when interacting with the outside world (incoming network responses, user input, etc).

For example, maybe you have a function that takes a number but it has to be a positive number. You can restrict this at a type level:

```ts
// positiveNumber.ts
declare const brand: unique symbol;

type PositiveNumber<T extends number> = T & {
  [brand]: "PositiveNumber";
};

export function isPositiveNumber<T>(number: T): number is PositiveNumber {
  return number >= 0
}
```

```ts
function foobar(n: PositiveNumber) { ... }
```

Rather than having to throw or return an error inside _foobar_ you now let the consumer of foobar deal with the fact that they can't just pass any number. If the consumer has a value of of type _number_ and they'd like to use _foobar_ they will have to make a check themselves:

```ts
// consumer.ts
function myFunc(n: number) {
  if (isPositiveNumber(n) {
     
  }
```

Reply:
Thanks for taking the time to write that up. Very cool. Thanks!

Reply:
Here's a nice video on the concept as well: https://www.youtube.com/watch?v=rpw59rajUSI